### ISSUE TITLE
Feature request: Mask union

### ISSUE THREAD

**Issue by Pr0methean** [RazrFalcon/tiny-skia#76]
I have an app where I load layers from SVGs, repaint them various colors, stack them, and output the results as PNGs. To minimize the memory consumed by intermediate results, I always union together masks that are being painted the same color for the same output PNG, unless I need to avoid doing so for Z-order reasons. Thus, being able to take the union of two masks via a tiny-skia routine rather than a loop I had to write myself, would be very useful.

As well, I could get by with fewer SVGs given subtraction and XOR operations on masks.

**Comment by RazrFalcon** [comment]
Yes, the `Mask` API is very primitive right now. Will add more operations.

**Comment by Pr0methean** [comment]
Another useful operation would be to multiply all pixels in an alpha mask by a fraction.

**Comment by RazrFalcon** [comment]
Can you explain the use case? And it would be easier to implement on your side via `data_mut`. I don't think SIMD would help here much.

**Comment by Pr0methean** [comment]
The use case would involve merging consecutive layers that have the same color but different opacity, so that the layers can be masks rather than pixmaps until they've been combined.

Also, would SIMD help at all in stacking two masks where one has the same alpha for the whole image? I'd use this to make stained-glass window textures (where the background is the same color as the frame but 75% transparent).

**Comment by Pr0methean** [comment]
And is there any chance that once `from_pixmap` is optimized, a version can be provided that would replace the contents of a mutably-borrowed Mask rather than create a new one? I'd use that with an object pool.

**Comment by RazrFalcon** [comment]
Will take a look.

>a version can be provided that would replace the contents of a mutably-borrowed Mask rather than create a new one?

You mean something like `Pixmap::to_mask(&self, mask_type: MaskType, mask: &mut Mask)`?
Yes, reducing allocations the goal. 

The current `tiny-skia` API is incredibly bare-bone, mainly because it contains only stuff I use in [resvg](https://github.com/RazrFalcon/resvg). So any suggestions are welcome, as long as I have time implementing them.
